src/main/java/com/iridium/iridiumskyblock/gui/IslandPermissionsGUI.java

```diff
@@ -57,17 +62,28 @@ public void addContent(Inventory inventory) {
     public void onInventoryClick(InventoryClickEvent event) {
         for (Map.Entry<String, Permission> permission : IridiumSkyblock.getInstance().getPermissionList().entrySet()) {
             if (permission.getValue().getItem().slot != event.getSlot()) continue;
+            if (permission.getValue().getPage() != page) continue;
 
             User user = IridiumSkyblock.getInstance().getUserManager().getUser((Player) event.getWhoClicked());
             if (user.getIslandRank().getLevel() <= islandRank.getLevel() || !IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(getIsland(), user, PermissionType.CHANGE_PERMISSIONS)) {
                 event.getWhoClicked().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotChangePermissions.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
             } else {
                 boolean allowed = IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(getIsland(), islandRank, permission.getValue(), permission.getKey());
-                IridiumSkyblock.getInstance().getIslandManager().setIslandPermission(getIsland(), islandRank, permission.getValue(), permission.getKey(), !allowed);
+                IridiumSkyblock.getInstance().getIslandManager().setIslandPermission(getIsland(), islandRank, permission.getKey(), !allowed);
                 event.getWhoClicked().openInventory(getInventory());
             }
             return;
         }
+
+        if (event.getSlot() == getNoItemGUI().size - 7 && page > 1) {
+            page--;
+            event.getWhoClicked().openInventory(getInventory());
+            return;
+        }
+        if (event.getSlot() == getNoItemGUI().size - 3) {
+            page++;
+            event.getWhoClicked().openInventory(getInventory());
+        }
```

Comment:
Also please introduce a variable for `event.getWhoClicked()`, you use it so often...
I just added that empty line in the code below.
 ```suggestion
        if (event.getSlot() == getNoItemGUI().size - 7 && page > 1) {
            page--;
            event.getWhoClicked().openInventory(getInventory());
            return;
        }
        
        if (event.getSlot() == getNoItemGUI().size - 3) {
            page++;
            event.getWhoClicked().openInventory(getInventory());
        }
```